common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -798,7 +798,10 @@ class @Problem
     #   'changeText' is a boolean to determine if there is need to change the
     #    text of submit button as well.
     attempts_remaining = @submitButton.data('attempts-remaining')
-    if enable && attempts_remaining != 0
+    if enable && attempts_remaining == 0
```

Comment:
Can you rewrite this as below? I believe the logic is the same, and I find it easier to understand.

If I've screwed up the logic, ignore this suggestion!

````
    if enable
      if attempts_remaining != 0
        @submitButton.removeAttr 'disabled'
        # the is-disabled is needed for Lettuce test below
        # lms/djangoapps/courseware/features/problems.py submit_problem(step)
        @submitButton.removeClass 'is-disabled'
      if changeText
        @submitButtonLabel.text(@submitButtonSubmitText)
    else
      @submitButton.attr({'disabled': 'disabled'})
      ....